indsl/statistics/outliers.py

```diff
@@ -219,55 +219,66 @@ def _get_outlier_indices(
         raise UserValueError("eps should be > 0.0.")
 
     df = data.to_frame()
+    df = (
+        df.dropna()
+    )  # dropping nan values, as nearest neighbors and other regression functions do not handle nan values
     df = df.rename(columns={df.columns[0]: "val"})
-
     df["rolling_mean"] = df["val"].rolling(time_window, min_periods=1).mean()  # calculate features
 
     if del_zero_val:  # delete 0 values if user requests it
         df = df[df["val"] != 0.0]
 
     df_dbscan = df[["val"]].copy()
     validate_series_is_not_empty(df_dbscan)
-
-    if eps is None:
+    if eps is None:  # If eps is not given by user, calculate best eps to use for dbscan with nearest neighbors
         # calculate distance to the further neighbor in order to find best epsilon (radius) parameter for dbscan
-        n_neighbors = min(len(df) - 3, 6)
+        n_neighbors = min(max(1, len(df) - 3), 6)
         nbrs = NearestNeighbors(n_neighbors=n_neighbors).fit(df)
         distances, _ = nbrs.kneighbors(df)
         df["knn_dist"] = distances[:, -1]
 
         dist = np.sort(distances[:, -1])
         i = np.arange(len(dist))
-        knee = KneeLocator(i, dist, S=1, curve="convex", direction="increasing", interp_method="polynomial")
-        eps = dist[knee.knee] if dist[knee.knee] > 0 else 0.5  # if calculated eps is 0, use 0.5 as default value
+        try:  # handle cases where np regressor of KneeLocator does not converge
+            knee = KneeLocator(i, dist, S=1, curve="convex", direction="increasing", interp_method="polynomial")
```

Comment:
can this fail? I assume that this instantiation should be succeeded.

Reply:
@tuanng-cognite 

With input data: input = [0, 9, 0, 0, 0, 0] it fails with:
  File "/Users/janne/Library/Caches/pypoetry/virtualenvs/indsl-Ir9U8kHg-py3.11/lib/python3.11/site-packages/numpy/linalg/linalg.py", line 124, in _raise_linalgerror_lstsq
    raise LinAlgError("SVD did not converge in Linear Least Squares")
numpy.linalg.LinAlgError: SVD did not converge in Linear Least Squares

With input data: input = [0, 9, 0, 0, 6, 0, 0] it fails with:
  File "/Users/janne/PycharmProjects/Product/indsl/indsl/statistics/outliers.py", line 245, in _get_outlier_indices
    eps = (dist[knee.knee] if dist[knee.knee] > 0 else 0.5)  # if calculated eps is 0, use 0.5 as default value
                              ^^^^^^^^^^^^^^^^^^^
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()

Reply:
But the way to handle this best might be to valide a certain length etc. of the inputs i and dist to the KneeLocator function, however I was not able to find much information about when this function will not converge/edge cases/inputs it cannot handle... 🤔

Reply:
It is obviously also not the most sensible input data (as 0 values are set to be removed), but I am assuming that we need to consider that the user can send in 'anything'?

Reply:
with arbitrary inputs, invalid value can happen. However, our logic is
```
dist = np.sort(distances[:, -1])
 i = np.arange(len(dist))
 KneeLocator(i, dist, S=1, curve="convex", direction="increasing", interp_method="polynomial")
 ```
It should be always valid if I understand correctly as `i` and `dist` are increasing arrays.

Reply:
With the different input I tested, the code failed due to the mentioned errors (see the test, these are not input directly into the KneeLocator, but to the _outlier_indices() function. So, as far as I see the question is then if we should have some validation on the length/shape of dist and i, or use a try, except. What do you think, @tuanng-cognite ?

Reply:
`[0, 9, 0, 0, 0, 0]` as input, I see the `i` and `dist` has length 1. How about testing the length of `dist`, if <= 1, eps = 0.5 else 
```
knee = KneeLocator...
...
```

Reply:
@tuanng-cognite Checking for len(dist) > 2, as it does not really make sense to calculate a knee on a straight line with 2 datapoints